### ISSUE TITLE
Add a camera

### ISSUE THREAD

**Issue by mlms13** [fponticelli/ledge#4]
**Disclaimer:** I'm basing all of this on internet answers that are 2+ years old. I haven't done very thorough research to see if things have changed in Pixi since then.
### Old Info for Pixi Cameras

You can create [camera-like functionality in Pixi](https://github.com/GoodBoyDigital/pixi.js/issues/83#issuecomment-16857614) using `PIXI.DisplayObjectContainer` (which I think is [pixi.core.display.Container](https://github.com/pixijs/pixi-haxe/blob/master/pixi/core/display/Container.hx) in the Haxe externs).

You create one Container for the camera (with a width and height that will match the size of the canvas) and a second Container for the world, which should have width and height >= the width and height of the camera. The camera never actually moves, instead you offset the world Container so that different sections move into and out of view.

Most positioned things in the game are attached to the "world" Container, but HUD-related things (mini-map, timeline, other fixed controls) are attached directly to the "camera" Container so that they don't move when the world Container's offset is changed.
### Cameras in ECS

Next, we have to figure out how this fits into an ecs structure. Maybe we create a `CameraPosition` component that adds things to the camera Container, while the normal `Position` component adds things to the world. All of the positioning of the camera would be controlled by keyboard and/or mouse-related systems.

That might be enough to get the basics working, but we still have to figure out how to handle more advanced camera features like rotation and zoom. [This Stack Exchange question](http://gamedev.stackexchange.com/questions/32693/implementing-features-in-an-entity-system) has some useful answers. One suggests making the camera an entity with its own Rotation and Scale components; another suggests making the camera a component with a depth property and isRendering flag.

In conclusion, we'll have to try it out and see what works.
